dts/arm/raspberrypi/rpi_pico/rp2040.dtsi

```diff
@@ -10,9 +10,12 @@
 #include <zephyr/dt-bindings/clock/rpi_pico_clock.h>
 #include <zephyr/dt-bindings/i2c/i2c.h>
 #include <zephyr/dt-bindings/regulator/rpi_pico.h>
+#include <zephyr/dt-bindings/reset/rp2040_reset.h>
 #include <mem.h>
 
-#include "rpi_pico_common.dtsi"
+#ifndef RPI_PICO_DEFAULT_IRQ_PRIORITY
+#define RPI_PICO_DEFAULT_IRQ_PRIORITY 3
+#endif
```

Comment:
What is this? Why is this here?

Reply:
> Why is this here?

This is unchanged from the current `HEAD` of `main`. It also reflects the same usage pattern found in [dts/common/nordic/nrf_common.dtsi#L22](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/common/nordic/nrf_common.dtsi#L22)

Reply:
I don't see any benefit to having it here. It would be easier to manage if it was in Kconfig.

Reply:
I don't understand how that would work/be implemented. Can you explain? If this is a blocker to the PR, then someone else will need to raise a separate ticket/issue and get a PR merged into `main` which I can rebase on.

My understanding is that device trees(/fragments) are meant to be independent/orthogonal to their use (e.g. Linux kernel, Zephyr), and I can't find any example of `CONFIG_*` used directly in any `*.dts*` file.

Reply:
We should not use Kconfig symbols in DTS. We also should not use `#define`. Instead, include a `dt-bindings` header including the definitions.

Reply:
Pretty sure you cannot use Kconfig symbol in dts, the devicetree is used to generate Kconfig symbols in the first place and it is parsed before the whole Kconfig stuff. But you could certainly remove a property and use the Kconfig symbol in the code directly.

Reply:
Thanks, @henrikbrixandersen . I've created #79719 to capture this.

I would like this not to be a blocker for this PR being accepted. This PR is already quite big, in my opinion. If it is decided it's a blocker (at least for everything under Raspberry Pi), then who makes that decision? At the moment there are many reviewers who have either already previously approved this PR, or who are yet to provide their feedback. 

@kartben  , as the developer advocate, I'd really like your comment here. I don't really have the time to keep widening the scope of this PR.

Reply:
> then who makes that decision

I'd say @yonsch, the pico platform maintainer.

Reply:
This took over from the original code, and since an issue has already been created to handle this, I think it's okay to leave.

Reply:
Needs fixing but can be fixed in a later PR since this inherits what's already in tree

Reply:
Can this conversation be marked as "resolved"?